runtime/rasdump/trigger.c

```diff
@@ -555,7 +556,61 @@ prepareForDump(struct J9JavaVM *vm, struct J9RASdumpAgent *agent, struct J9RASdu
 		newState |= J9RAS_DUMP_TRACE_DISABLED;
 	}
 
-	if ((context->eventFlags & (J9RAS_DUMP_ON_GP_FAULT | J9RAS_DUMP_ON_ABORT_SIGNAL | J9RAS_DUMP_ON_TRACE_ASSERT)) == 0) {
+	/* Release vm access until this thread has the dumpKey and is ready to run. This will allow other threads to obtain exclusiveVMAccess in the meantime. */
+	if (vmThread) {
+		if (J9_ARE_ANY_BITS_SET(vmThread->publicFlags, J9_PUBLIC_FLAGS_VM_ACCESS)) {
```

Comment:
This is still not correct - please follow the pattern below.

Reply:
Strictly speaking, you don't need to do anything if inNative is true - the thread will not block exclusive. That should simplify the code.
```
if (!inNative) {
   if (vm access) {
      release and set your variable
   }
}
```

Reply:
And when reacquiring access, you need only check the local variable and call acquire - no need for any enter/exit VM in this change.

It's safe to check `!inNative` without an ifdef, as the field always exists and will always be 0 in VMs which have not been updated for atomic-free JNI.

Reply:
Thanks for the review @gacholio I've updated the change with the pattern you suggested.